### ISSUE TITLE
Solution for frequent “Connection Failed” (Caused by Shift key triggering AI Notes / AI Preview)

### ISSUE THREAD

**Issue by artsparkAI** [cursor/cursor#1783]
TLDR
If you:

Enabled AI-Preview
Have custom API keys
Accidentally held shift too long while coding
You’ll get Connection Failed popups a lot. You can solve it by turning off custom API Keys or by turning off AI-Preview feature.

Hi everyone, you may have noticed a frequent “Connection Failed” popup when using Cursor.

First of all, if you didn’t know this is actually caused by holding down the Shift key triggering the AI Notes / AI Preview feature (which tries to search for info for the symbol currently selected).

This is super confusing because holding the shift key too long is very easy to do in day to day editing. Perhaps the shortcut should be changed?

Second of all, the true cause of this bug is because AI-Notes feature DOES NOT support custom API keys for the model. If you go to your settings and turn off custom models, you’ll find that AI-Notes works (It’s a great feature btw) and you no longer get the Connection Failed pop-up

edit:  Seems to be specifically Anthropic custom API key that causes the issue, not OpenAI

[](https://forum.cursor.com/t/solution-for-frequent-connection-failed-caused-by-shift-key-triggering-ai-notes-ai-preview/14070?u=artspark)

**Comment by tjx666** [comment]
I didn't use custom key, also didn't see the popup

**Comment by leerob** [comment]
Thanks for opening this issue! We've moved our community support to the [Cursor Forum](https://forum.cursor.com) where you can get help from both the team and other users.

If this is a specific bug that you can still reproduce, please open a new issue with clear reproduction steps on the forum.

Thanks you!
